src/engine/strat_engine/liminal/liminal.rs

```diff
@@ -190,47 +212,49 @@ impl LiminalDevices {
             (Err(e), _) => return Err(e),
         };
 
+        let uuids = unlocked_devices
```

Comment:
I think `unzip()` may actually be less performant because we would actually have to collect the discarded elements into a `Vec` and then immediately drop it. At the very least, this is an unnecessary allocation.

Reply:
ok!